ami/models/components/unet/middle_block.py

```diff
@@ -0,0 +1,63 @@
+# Ref: https://github.com/facebookresearch/RCDM
+
+
+import torch
+import torch.nn as nn
+
+from .commons import AttentionBlock, ResBlock
+
+
+class UnetMiddleBlock(nn.Module):
+    """A block between encoder and decoder in Unet."""
+
+    def __init__(
+        self,
+        in_and_out_channels: int,
+        timestep_embed_dim: int,
+        num_heads: int,
+    ) -> None:
+        """A block between encoder and decoder in Unet.
+
+        Args:
+            in_and_out_channels (int):
+                Input and output num of channels.
+            timestep_embed_dim (int):
+                Dims of timestep embedding.
+            num_heads (int):
+                num_head when adding attention layers.
+        """
+        super().__init__()
+
+        self.resnet_blocks = nn.ModuleList(
+            [
+                ResBlock(
+                    in_channels=in_and_out_channels, emb_channels=timestep_embed_dim, out_channels=in_and_out_channels
+                ),
+                ResBlock(
+                    in_channels=in_and_out_channels, emb_channels=timestep_embed_dim, out_channels=in_and_out_channels
+                ),
+            ]
+        )
+        self.attentions = nn.ModuleList([AttentionBlock(in_and_out_channels, num_heads=num_heads)])
```

Comment:
わざわざModuleListにする必要がない。
```suggestion
        self.resblock_in = ResBlock(
                    in_channels=in_and_out_channels, emb_channels=timestep_embed_dim, out_channels=in_and_out_channels
                )
        self.resblock_out = ResBlock(
                    in_channels=in_and_out_channels, emb_channels=timestep_embed_dim, out_channels=in_and_out_channels
                )
        self.attention = AttentionBlock(in_and_out_channels, num_heads=num_heads)]
```